### PR TITLE
[merged] Add HostCreds resource.

### PR DESCRIPTION
**Pull request by ashcrow**
This change presents a simple API for https://github.com/projectatomic/commctl/pull/17 to use along with it's tests and documentations.

### CODE REVIEW

File: src/commissaire/handlers/hosts.py [hunk 1]
```diff
@@ -57,6 +57,39 @@ def on_get(self, req, resp):
...
+        :type req: falcon.Request
+        :param resp: Response instance that will be passed through.
+        :type resp: falcon.Response
+        :param address: The address of the Host being requested.
+        :type address: str
+        """
+        # TODO: Verify input
+        # TODO: Decide if this should be a model or if it makes sense to
+        #       stay a subset off of Host and bypass the req.context
+        #       middleware system.
```
Comment (mbarnes) [inline src/commissaire/handlers/hosts.py]: I'd say nah, using the `Host` model is fine here.

**Comment by mbarnes** [comment]
LGTM.  I'll give @cooktheryan a chance to comment.

**Comment by ashcrow** [comment]
@cooktheryan it's up to you if you'd like to review it or if you want to defer to @mbarnes review.

**Comment by cooktheryan** [comment]
:thumbsup:  I like it

**Comment by cooktheryan** [comment]
@mbarnes if you want to review it go for it. I really like the concept of it but just dont have time to review the code and functionality today

**Comment by ashcrow** [comment]
@mbarnes :arrow_up:

File: doc/endpoints.rst [hunk 2]
```diff
@@ -322,6 +322,22 @@ DELETE
...
+
+GET
+```
+Retrieve a specific hosts credentials.
+
+.. code-block:: javascript
+
+   {
+       "ssh_priv_key": string,       // base64 encoded ssh private key
+       "remote_user":  enum(string), // name of ssh user to use for connections
```
Comment (mbarnes) [inline doc/endpoints.rst]: Tiny nit: shouldn't this just be `string`?  I don't get what `enum(string)` means.

**Comment by mbarnes** [comment]
Tiniest little nit, but otherwise I'd say good to go.

File: doc/endpoints.rst (hunk 2 above)
Comment (ashcrow) [inline doc/endpoints.rst]: You're right. It should be string. There isn't an enumeration for users. Fixing.

**Comment by mbarnes** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**